api/internal/git/repospec.go

```diff
@@ -238,19 +244,20 @@ func extractHost(n string) (string, string) {
 	// Validate the username and scheme before attempting host/path parsing, because if the parsing
 	// so far has not succeeded, we will not be able to extract the host and path correctly.
 	if err := validateScheme(scheme, acceptSCP); err != nil {
-		// TODO: return this error instead.
-		return "", n
+		return "", "", err
 	}
 
 	// Now that we have extracted a valid scheme+username, we can parse host itself.
 
 	// The file protocol specifies an absolute path to a local git repo.
 	// Everything after the scheme (including any 'username' we found) is actually part of that path.
 	if scheme == "file://" {
-		return scheme, username + n
+		return scheme, username + n, nil
+	}
+	var host, rest = n, ""
+	if sepIndex := findPathSeparator(n, acceptSCP); sepIndex > 0 {
+		host, rest = n[:sepIndex+1], n[sepIndex+1:]
```

Comment:
Before, when we didn't find a separator, the whole thing was considered a path. Now it's considered a host. I think this makes more sense, and it maintains the current error messages in some cases. Notably, `ssh://git.example.com` should complain about orgRepo, not host, missing. Previously all parsing would complete before any errors would return, and for whatever reason orgRepo was checked first, resulting in the correct message for that case despite host also being empty.